src/components/clusterConfiguration/discoveryTroubleshootingModal.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  Button,
+  Modal,
+  ButtonVariant,
+  ModalVariant,
+  TextContent,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import { ToolbarButton } from '../ui/Toolbar';
+import PrismCode from '../ui/PrismCode';
+
+type DiscoveryTroubleshootingModalButtonProps = React.ComponentProps<typeof Button> & {
+  ButtonComponent?: typeof Button | typeof ToolbarButton;
+  onClick?: () => void;
+};
+
+export const DiscoveryTroubleshootingModalButton: React.FC<DiscoveryTroubleshootingModalButtonProps> = ({
+  ButtonComponent = Button,
+  onClick,
+  children,
+  ...props
+}) => {
+  const [isModalOpen, setIsModalOpen] = React.useState(false);
+  const closeModal = () => setIsModalOpen(false);
+  const handleClick = onClick || (() => setIsModalOpen(true));
+  return (
+    <>
+      <ButtonComponent variant={ButtonVariant.link} {...props} onClick={handleClick} isInline>
+        {children}
+      </ButtonComponent>
+      <DiscoveryTroubleshootingModal isOpen={isModalOpen} onClose={closeModal} />
+    </>
+  );
+};
+
+type DiscoveryTroubleshootingModalProps = {
+  onClose: () => void;
+  isOpen: boolean;
+};
+
+export const DiscoveryTroubleshootingModal: React.FC<DiscoveryTroubleshootingModalProps> = ({
+  onClose,
+  isOpen,
+}) => {
+  return (
+    <Modal
+      title="Host Discovery Troubleshooting"
+      isOpen={isOpen}
+      actions={[
+        <Button key="close" variant={ButtonVariant.primary} onClick={onClose}>
+          Close
+        </Button>,
+      ]}
+      onClose={onClose}
+      variant={ModalVariant.large}
+    >
+      <TextContent>
+        <Text component={TextVariants.p}>
+          Booting up an ISO might take a few minutes, and might depend heavily on your hardware and
+          network configuration, however, if after a few minutes your machine does not show up in
+          the UI, it's probably best to try and login to it via SSH, for that you would need to know
+          the machine ip-address, or use a console of some sort.
+        </Text>
+        <Text component={TextVariants.h2}>SSH into your machine</Text>
+        <PrismCode code={`ssh core@<machine-ip>`} />
+        <Text component={TextVariants.p}>
+          If SSH key was provided when generating the ISO it is not required to provide a password.
+        </Text>
+        <Text component={TextVariants.h2}>
+          Verify that the discovery agent is running correctly
+        </Text>
+        <Text component={TextVariants.p}>Execute</Text>
+        <PrismCode code={`ps -ef | grep agent`} />
+        <Text component={TextVariants.p}>
+          and see whether the agent runs with correct parameters. The result should look similar to
+          this:
+        </Text>
+        <PrismCode
+          code={`root        1342       1  0 09:56 ?        00:00:00 /usr/local/bin/agent --host 192.168.39.162 --port 30956 --cluster-id a8142e14-8bfe-46bd-bfbd-4bd1a126892b core        2063    2003  0 10:41 pts/0    00:00:00 grep --color=auto agent`}
```

Comment:
maybe extract the PrismCode `code` value so it will be easier to read ?
```suggestion
          code={PRISM_VERIFY_AGENT}
```

Reply:
I've avoided doing this as it would IMHO make editing of that documentation harder